app/lib/search/backend.dart

```diff
@@ -138,6 +147,71 @@ class SearchBackend {
   }
 }
 
+class PopularityStorage {
+  final Storage storage;
+  final Bucket bucket;
+
+  final Map<String, double> _values = {};
+
+  PopularityStorage(this.storage, this.bucket);
+
+  double operator [](String package) => _values[package];
+
+  Future init() async {
+    await fetch();
+    new Timer.periodic(const Duration(days: 1), (_) {
+      fetch();
+    });
+  }
+
+  Future fetch() async {
+    _logger.info('Updating popularity storage...');
+    final List<BucketEntry> list = await bucket.list().toList();
+    final List<String> names = list
+        .where((entry) => entry.isObject)
+        .map((entry) => entry.name)
+        .toList();
+    if (names.isEmpty) {
+      _logger.warning('No popularity data found!');
+      return;
+    }
+    // Try to load the available snapshots in reverse order (latest first).
+    names.sort();
+    for (String selected in names.reversed) {
+      _logger.info('Loading popularity data: $selected');
+      try {
+        final String json =
+            await bucket.read(selected).transform(UTF8.decoder).join();
+        final Map map = JSON.decode(json);
+        _updateLatest(map);
+        _logger.info('Popularity data loaded successful: $selected');
+        return;
+      } catch (e, st) {
+        _logger.severe('Unable to load popularity data: $selected', e, st);
+      }
+    }
+    return;
+  }
+
+  void _updateLatest(Map raw) {
+    final Map<String, int> rawTotals = {};
+    final List items = raw['items'];
+    for (Map item in items) {
+      final String package = item['pkg'];
+      final int devVotes = int.parse(item['votes_dev']);
+      final int directVotes = int.parse(item['votes_direct']);
+      final int totalVotes = int.parse(item['votes_total']);
+      final int finalVotes = directVotes * 25 + devVotes * 5 + totalVotes;
+      rawTotals[package] = finalVotes;
+    }
+    final summary = new Summary(rawTotals.values);
+    for (String package in rawTotals.keys) {
+      final int raw = rawTotals[package];
+      _values[package] = summary.bezierScore(raw);
```

Comment:
There is still no document explaining all the scoring logic and furthermore no clear motivation what benefit (or drawbacks) this non-linear scaling has, except for: "npmjs does it, so let's hope they had a good reason, the same reason applies to us and therefore let's just copy it".

Reply:
Yeah, I'll make it a priority to start a doc on it before the end of this week.

For the score value in this code, we need a normalization, and bezier is just as good as anything else for now. (I believe that once we have the building blocks in place, changing the normalization or the weights, or in general figuring out what we really want with the metrics will much easier with live data at hand.)